### PR TITLE
Updates Hydrogen to 2.0.5

### PR DESCRIPTION
**Pull request by substrae**
## ✨ Summary

This pull request updates Hydrogen through [`2.0.4`](https://github.com/hydrogen-css/hydrogen/releases/tag/v2.0.4) to [`2.0.5`](https://github.com/hydrogen-css/hydrogen/releases/tag/v2.0.5) and includes the following changes:
* Migration from PostCSS/CSSnano/Autoprefixer to Lightning CSS
* A reduction (half!) in compile time
* The ability to configure `font_weight` values in `hydrogen.config.json`
* More accurate CSS selector validation
* A fix for old references to `clean.js`
* A fix for broken `generate_logs` configuration settings
* Fixes to various invalid CSS declarations in the app that existed prior to the update

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Grab the branch
2. Rerun `pnpm`'s installation (you might need to delete your `node_modules`)
3. Ensure you've got `2.0.5` installed
4. Run `pnpm build` to confirm the build compiles successfully
5. Check for visual diffs specific to the fixed bugged CSS declarations

### CODE REVIEW

File: apps/web/src/components/Table/ResponsiveTable/Table.tsx
Comment (mnigh) [inline apps/web/src/components/Table/ResponsiveTable/Table.tsx:100]: This doesn't render as it did before on the tables. The [chromatic changeset](https://www.chromatic.com/review?appId=61e099a1bb1465003a73d3ce&number=9793&type=linked&view=changes) on this PR also caught the differences.

## Screenshot

Screen Shot 2024-03-19 at 13 40 38

**Review by esizer — COMMENTED** [review]
I noticed a few things while testing this:

1. While hydrogen build times got cut nearly in half, the entire build time rose :thinking: 
2. The diff looks funny to me. Not sure if we meant to switch from row striping to column striping

**Review by mnigh — CHANGES_REQUESTED** [review]
The `Table` component css change will to be fixed and another run of Chromatic GitHub action to verify there are no changes before merging. Otherwise, 👍.

## Good news
Hydrogen builds 1 second faster 🥳!

<details><summary>Screenshots</summary>
Screen Shot 2024-03-19 at 13 23 09
Screen Shot 2024-03-19 at 13 22 49
</details>

## Not so great news
Overall build is 14 seconds slower (not sure why) 🤷. 

<details><summary>Screenshots</summary>
Screen Shot 2024-03-19 at 13 14 54
Screen Shot 2024-03-19 at 13 20 04
</details>

**Comment by substrae** [comment]
@esizer @mnigh 

Good catches on the Chromatic diffs - I was waiting on them to check them myself but got pulled away. 

In terms of the build time, wth? Can y'all confirm that the actual H2 build isn't bloating it somewhere?

**Comment by substrae** [comment]
Also, does the build time reflect the installation times for missing modules now that we're with PNPM? I wonder if subsequent runs will be shorter if that's the case.

**Comment by esizer** [comment]
> In terms of the build time, wth? Can y'all confirm that the actual H2 build isn't bloating it somewhere?

I'm not too sure, the only thing that should have changed between this branch and main is hydrogen. Is there anything happening outside of the calculated build time we see in the CLI output?

> Also, does the build time reflect the installation times for missing modules now that we're with PNPM? I wonder if subsequent runs will be shorter if that's the case.

The numbers I posted were just build times, it did not include any module installation. Everything else seemed to be the same. The onyl difference I noticed was the hydrogen css file was slightly larger but I assume that is explained by the rules that were fixed :thinking:

**Comment by yonikid15** [comment]
I'm seeing similar results. Not sure what would cause it to slow down.
(I think it's time for a new laptop 😅)


```sh
# Before (main)
Tasks:    22 successful, 22 total
Cached:    1 cached, 22 total
  Time:    6m28.927s
  
# After (this branch)
   Tasks:    22 successful, 22 total
Cached:    0 cached, 22 total
  Time:    8m39.563s
```

**Comment by vd1992** [comment]
Main `pnpm run dev:fresh` in minutes
3:54 
3:43
3:40

Branch
4:41
4:40
WSL restarted
3:47
3:45

No difference for me, I think I have an unrelated WSL issue with it being a CPU hog sometimes needing a restart. Not sure where or when that became a thing.

**Comment by substrae** [comment]
Oof, this feels like it's getting even more mysterious the more people chime in 😂 

@esizer Assuming you don't have the same problems I do with `node_modules` permissions, you slap a timer at the start and end of the build script if you want to see if there's any time being excluded from the build... but I'm pretty sure the timer starts and stops there already: https://github.com/hydrogen-css/hydrogen/blob/main/lib/build.js

**Comment by esizer** [comment]
> Oof, this feels like it's getting even more mysterious the more people chime in 😂
> 
> @esizer Assuming you don't have the same problems I do with `node_modules` permissions, you can slap a timer at the start and end of the build script if you want to see if there's any time being excluded from the build... but I'm pretty sure the timer starts and stops there already: https://github.com/hydrogen-css/hydrogen/blob/main/lib/build.js

I might need to get someone else to test but I just did a few different tests and it is honestly like a game of chance for me. I swtich back and forth between main and this branch and sometimes main is faster and sometimes this branch is faster :woman_shrugging: 

It is hard to tell for me since the times are only differing by seconds (not mins like for @yonikid15 :laughing: )

**Comment by substrae** [comment]
Assuming someone can confirm the same time variations, I'm inclined to suggest it's not Hydrogen related and likely has to do with migrating to PNPM...

**Review by mnigh — APPROVED** [review]
The `Table` component css change worked and shows no diffs in Chromatic 👍

**Comment by vd1992** [comment]
Tested some more, build times hardly different, in fact in some runs it is slightly faster 🤷 All good I guess?

**Comment by mnigh** [comment]
> Tested some more, build times hardly different, in fact in some runs it is slightly faster 🤷 All good I guess?

@substrae are you waiting on anything else from any of us? if not, can you merge this to the `main` branch?

**Pull request merged**
